.github/workflows/ci.yaml

```diff
@@ -0,0 +1,48 @@
+name: Run codebase checks and tests
+
+on: [push, pull_request]
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+
+    services:
+      db:
+        image: postgres:11
+        ports: ["5432:5432"]
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    env:
+      MIX_ENV: test
+      DATABASE_URL: postgres://postgres:postgres@localhost/elixir_boilerplate_test
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-elixir@v1
+        with:
+          elixir-version: 1.10.1
+          otp-version: 22.2 
+      - uses: actions/cache@v1
+        id: cache-mix
+        with:
+          path: deps
+          key: ${{ runner.os }}-mix-${{ hashFiles(format('{0}/mix.lock', github.workspace)) }}
+          restore-keys: |
+            ${{ runner.os }}-mix-
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12.15
+      - uses: actions/cache@v1
+        id: cache-npm
+        with:
+          path: ~/.npm
```

Comment:
I had better success speed wise caching the node_modules directory in my tests, but i don't know how that might turn bad. It was also using yarn, maybe npm handles its cache folder in a better way. But you should try and compare! :)

Reply:
Following `actions/cache@v1` documentation:

> For npm, cache files are stored in ~/.npm on Posix, or %AppData%/npm-cache on Windows. See https://docs.npmjs.com/cli/cache#cache
>
> > _Note_: It is not recommended to cache `node_modules`, as it can break across Node versions and won't work with `npm ci`

https://github.com/actions/cache/blob/master/examples.md#node---npm